ykrt/src/mt.rs

```diff
@@ -347,12 +355,76 @@ impl MTThread {
                     }
                 }
             }
-            PHASE_TRACING => {
+            PHASE_TRACING | PHASE_TRACING_LOCK => {
```

Comment:
Wouldn't it be easier to split this out into separate cases? It seems the nested `PHASE_TRACING_LOCK` case doesn't re-use any code of the `PHASE_TRACING` case. The former always returns before getting there.

Reply:
Sorry, that's not true.

Reply:
I think you've realised this, but the alternative would have to be a loop around the whole PHASE `match` statement, which I think would be harder to understand.

Reply:
I'm actually really struggling in general. See below.